enhancements/ingress/aws-load-balancer-operator.md

```diff
@@ -317,43 +318,48 @@ does not remove the existing addon attachment from the provisioned load
+can be used to generate certificates and have them injected into the webhook
+configurations. The operator does not manage the lifecycle of the certificates
+in the associated secret. Additional details mentioned in [Drawbacks](#drawbacks).
 
 ### Risks and Mitigations
 
-#### Restricting features in the lb-controller
-
-Some changes will have to be made in the lb-controller to ensure that the users
-only use features which are supported on OCP and only those features which
-target Ingress resources. These changes may also be contributed back upstream
-since it may benefit other users.
-
-1. The annotation `alb.ingress.kubernetes.io/target-type` can be set to the
-value **_ip_** or **_instance_**. When the value is set to _instance_ the
-created load balancer will be configured to route traffic through a NodePort
-service. The _ip_ value only works when the cluster uses the Amazon VPC CNI and
-hence will not work on OCP/OKD clusters. The controller has to be modified so
-that this annotation is ignored and the controller always uses routing through
-_NodePort_ services.
-2. The _lb-controller_ has support for _Service_ type resources as
-well. When the annotation `service.beta.kubernetes.io/aws-load-balancer-type:
-"external"` is specified on a _Service_ resource of type _LoadBalancer_ the
-in-tree controller ignores this resource and the lb-controller instead
-provisions a Network Load Balancer with the correct configuration. Since we want
-to restrict the features supported by the controller to only Ingresses the
-lb-controller will have to be modified so that it does not reconcile any Service
-resources.
+The annotation `alb.ingress.kubernetes.io/target-type` can be set to the
+value **_ip_** or **_instance_**. When the value is set to _instance_ (default)
+the created load balancer will be configured to route traffic through a NodePort
+service. Setting the annotation to _ip_ is not huge risk, since it only works when
+the cluster uses the Amazon VPC CNI and hence will not work on OCP/OKD clusters.
+
+### Drawbacks
+
+Since we are reusing an existing upstream controller and restricting which of its
+features are enabled, some upstream documentation would not be applicable for this
+operator.
+
+The _lb-controller_ currently has support for _Service_ type resources as well.
+When the annotation `service.beta.kubernetes.io/aws-load-balancer-type: "external"`
+is specified on a _Service_ resource of type _LoadBalancer_ the in-tree controller
+ignores this resource and the lb-controller instead provisions a Network Load
+Balancer with the correct configuration.
+
+The operator utilises the `service-ca-controller` to generate and inject certificates
+using the associated secret. Currently upon renewal of these certificates, the operator
+does not reload the deployment. Dynamic reload of the deployment isn't supported in
+the initial release.
```

Comment:
This means that the webhook will fail and break your cluster if the certificate expires, right?  So that's a 2-year timebomb.  It's crucial we not lose track of that.

Reply:
Noted, will have a corresponding task in the backlog.

Reply:
This issue was already mitigated in the initial release. The controller uses webhook rotation [feature](https://github.com/kubernetes-sigs/controller-runtime/pull/421) provided by the container runtime. 

Also verified manually, 
```bash
{"level":"info","ts":1655811187.8988125,"logger":"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}
{"level":"info","ts":1655811187.898957,"logger":"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}
```